etna/ciim/client.py

```diff
@@ -1,85 +1,266 @@
+import enum
 import logging
 
+from typing import Optional
+
 import requests
 
-from .exceptions import ConnectionError, InvalidResponse, KongError, KubernetesError
+from .exceptions import ConnectionError, KongError
 
 logger = logging.getLogger(__name__)
 
 
+class Stream(str, enum.Enum):
+    """Options for restricting /search results to a given stream.
+
+    Evidential:
+        Catalogue data i.e: Records
+
+    interpretive:
+        Any content written _about_ evidential i.e: Research Guides, Blog posts
+    """
+
+    EVIDENTIAL = "evidential"
+    INTERPRETIVE = "interpretive"
+
+
+class Sort(str, enum.Enum):
+    """Options for sorting /search results by a given field."""
+
+    TITLE = "title"
+    DATE_CREATED = "date_created"
+
+
+class SortOrder(str, enum.Enum):
+    """Options for sort order for /search results."""
+
+    ASC = "asc"
+    DESC = "desc"
+
+
+class Template(str, enum.Enum):
+    """Include @template block to assist parsing/rendering of record data.
+
```

Comment:
If this is the case " to assist parsing/rendering"  why we are passing it to Kong ?

Reply:
The `@template` block is easier to parse than searching through nested collections for data (see the transform functions).

I've updated the comment to make it clear that Kong is including the `@template` block in its response.

Reply:
Thank you , but the question is why do we pass **@template** in the request ?

Reply:
We can request a specific `@template` for Kong to include in the response.

From the API spec:

```json
{
    "name": "template",
    "in": "query",
    "description": "Template to return",
    "required": false,
    "schema": {
        "type": "string",
        "enum": [
            "details",
            "results"
        ]
    }
},
```

It's worth noting this is Kong's concept of a template, not Django/Wagtail's.